steps/JamesIves/ghpages-deploy.dhall

```diff
@@ -0,0 +1,33 @@
+let Step = ../../schemas/Step.dhall
+
+let AuthSchema = ./AuthSchema.dhall
+
+let auth =
+      λ(authSchema : AuthSchema) →
+        merge
+          { accessToken = toMap
+              { ACCESS_TOKEN = "\${{ secrets.ACCESS_TOKEN }}" }
+          , githubToken = toMap
+              { GITHUB_TOKEN = "\${{ secrets.GITHUB_TOKEN }}" }
+          , ssh = toMap { SSH = "true" }
+          }
+          authSchema
+
+in  λ ( args
+      : { authSchema : AuthSchema
+        , branch : Text
+        , folder : Text
+        , opts : List { mapKey : Text, mapValue : Text }
```

Comment:
Hi @gvolpe sorry for the late reply! Yeah I think this looks ok!

Small snag when users don't have extra options; they need to pass the arg like this: 

```js
opts = []: List { mapKey : Text, mapValue : Text }
```

We can probably add a new type:

```js
// leaning towards this
opts = []: Options
// but we can also have
opts = None Options
// this would require 
options = Some (toMap { ... })
```

wdyt?

Reply:
LGTM @regadas :+1: , just pushed these changes :)